extensions/sql-database-projects/src/models/project.ts

```diff
@@ -263,11 +263,11 @@ export class Project {
 		switch (itemType) {
 			case templates.preDeployScript:
 				xmlTag = constants.PreDeploy;
-				this.preDeployScripts.push(fileEntry);
+				this.preDeployScripts.length === 0 ? this.preDeployScripts.push(fileEntry) : this.noneDeployScripts.push(fileEntry);
 				break;
 			case templates.postDeployScript:
 				xmlTag = constants.PostDeploy;
-				this.postDeployScripts.push(fileEntry);
+				this.postDeployScripts.length === 0 ? this.postDeployScripts.push(fileEntry) : this.noneDeployScripts.push(fileEntry);
```

Comment:
Thanks for catching this. 
I realized noneDeployScripts was getting populated properly while opening a project; but not while adding a new script.